### PR TITLE
Let `build_ext` inherit from `setuptools.Command`

### PR DESCRIPTION
**Pull request by BvB93**
The `build_ext` class was previously missing inheritance from `setuptools.Command`, and hence subtypes were considered invalid mapping values for the `cmdclass` parameter back in `setuptools.setup`.

Examples
----------
``` python
In [1]: from setuptools.command.build_ext import build_ext

In [2]: build_ext.mro()
Out[2]: 
[setuptools.command.build_ext.build_ext,
 distutils.command.build_ext.build_ext,
 setuptools.Command,
 distutils.cmd.Command,
 object]
```

### CODE REVIEW

File: stubs/setuptools/setuptools/command/build_ext.pyi [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+
 have_rtld: bool
 use_stubs: bool
 libtype: str
 
 def if_dl(s): ...
 def get_abi3_suffix(): ...
 
-class build_ext(_build_ext):
+class build_ext(_build_ext, Command):
```
Comment (JelleZijlstra) [inline stubs/setuptools/setuptools/command/build_ext.pyi:13]: This might need to be type ignored.

**Review by JelleZijlstra — COMMENTED** [review]
How does this work exactly? At runtime I just see `class build_ext(_build_ext):`; is there some monkeypatching going on?

File: stubs/setuptools/setuptools/command/build_ext.pyi (hunk 1 above)
Comment (BvB93) [reply stubs/setuptools/setuptools/command/build_ext.pyi]: Done as of How does this work exactly? At runtime I just see class build_ext(_build_ext):; is there some monkeypatching going on?.

**Comment by BvB93** [comment]
> How does this work exactly? At runtime I just see `class build_ext(_build_ext):`; is there some monkeypatching going on?

It's not exactly my area of expertise, but this seems to be the case. Based on the at the mro output `setuptools.Command` is injected as a baseclass for the distutils `build_ext` class.

**Comment by Akuli** [comment]
The mro depends on whether `distutils.command.build_ext` is imported before `setuptools`:

```python3
akuli@akuli-desktop:~$ python3
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import distutils.command.build_ext
>>> import setuptools.command.build_ext
>>> setuptools.command.build_ext.build_ext.mro()
[<class 'setuptools.command.build_ext.build_ext'>, <class 'distutils.command.build_ext.build_ext'>, <class 'distutils.cmd.Command'>, <class 'object'>]
>>> 
akuli@akuli-desktop:~$ python3
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import setuptools.command.build_ext
>>> setuptools.command.build_ext.build_ext.mro()
[<class 'setuptools.command.build_ext.build_ext'>, <class 'distutils.command.build_ext.build_ext'>, <class 'setuptools.Command'>, <class 'distutils.cmd.Command'>, <class 'object'>]
>>> 
```

What practical problem do specific `build_ext` typings solve? Do you need to import `build_ext` directly into your program and use some `setuptools.Command` functionality?

I think we should do the typing like it is if you don't import `distutils.command.build_ext`, because distutils is deprecated.

**Comment by BvB93** [comment]
> What practical problem do specific `build_ext` typings solve? Do you need to import `build_ext` directly into your program and use some `setuptools.Command` functionality?
> 
> I think we should do the typing like it is if you don't import `distutils.command.build_ext`, because distutils is deprecated.

The problem that prompted this PR was a false positive when passing custom a `cmdclass`, the latter expecting a `Mapping[str, type[setuptools.Command]]` subtype. As `setuptools.command.build_ext.build_ext` currently does not inherit from `setuptools.Command` in the stub file, my first reaction was to explicitly add it as a baseclass (which apparently may or may not be true during runtime as you pointed out). 

Another option here would be to widen the type of `cmdclass` from `Mapping[str, type[setuptools.Command]]` to `Mapping[str, type[distutils.core.Command]]`.

``` python
from setuptools import setup
from setuptools.command.build_ext import build_ext

class build_test_ext(build_ext):...

setup(
    name="test",
    cmdclass={"build_ext": build_test_ext},
)
```

**Comment by BvB93** [comment]
> Another option here would be to widen the type of `cmdclass` from `Mapping[str, type[setuptools.Command]]` to `Mapping[str, type[distutils.core.Command]]`.

For now I've created a PR with this alternative approach at https://github.com/python/typeshed/pull/7458. I don't have a particularly strong opinion about which PR I found more desirable, as long as the problem gets fixed.

**Comment by Akuli** [comment]
I merged the alternative PR.

**Pull request closed without merging**